trigger/cron/cron.go

```diff
@@ -91,17 +100,33 @@ func (s *Scheduler) run(ctx context.Context) error {
 			continue
 		}
 
-		// calculate the next execution date.
-		job.Prev = job.Next
-		job.Next = sched.Next(now).Unix()
-
 		logger := logrus.WithFields(
 			logrus.Fields{
 				"repo": job.RepoID,
 				"cron": job.ID,
 			},
 		)
 
+		// calculate the next execution date.
+		prev := time.Unix(job.Prev, 0)
+		next := sched.Next(prev)
+
+		//If past in time, retreive the next valid time until next time will be greater than now
+		for next.Before(now) {
+			prev = next
+			next = sched.Next(next)
+			// if next still under now this scheduled  will be skipped , else will be executed !
+			if next.Before(now) {
+				logger.Warnf("skipped cron job [%s]  scheduled in the past at : [%s]  NOW [%s]", job.Name, prev, now)
```

Comment:
Too much spaces?

Reply:
Hi @tboerger thank you for the quick reply. 
What exactly would you refer? Are you talking aboult spaces in the log message?

Reply:
Yes, just refering to the duplicated spaces within the log message, so really nothing critical.